File: deploy/check_server.js
const {Client} = require('ssh2');
const conn = new Client();
function execCmd(cmd, timeout) {
  timeout = timeout || 20;
  return new Promise(function(resolve, reject) {
    conn.exec(cmd, function(err, stream) {
      if(err) { reject(err); return; }
      var out = '', errOut = '';
      stream.on('data', function(d) { out += d.toString(); });
      stream.stderr.on('data', function(d) { errOut += d.toString(); });
      stream.on('close', function() { resolve(out.trim() || errOut.trim()); });
      setTimeout(function() { reject('timeout'); }, timeout * 1000);
    });
  });
}
async function main() {
  await new Promise(function(res, rej) {
    conn.on('ready', res);
    conn.on('error', rej);
    conn.connect({
      host: 'newbb.test.bangbangvip.com',
      port: 22,
      username: 'ubuntu',
      password: 'Newbang888',
      readyTimeout: 15000
    });
  });
  console.log('CONNECTED');

  var r = await execCmd('docker exec 6b099ed3-7175-4a78-91f4-44570c84ed27-backend printenv | grep DATABASE_URL');
  console.log('DB_URL: ' + r);

  r = await execCmd('docker exec 6b099ed3-7175-4a78-91f4-44570c84ed27-backend cat /app/BUILD_INFO 2>/dev/null');
  console.log('Backend BUILD: ' + (r || 'NONE'));

  r = await execCmd('docker exec 6b099ed3-7175-4a78-91f4-44570c84ed27-admin cat /app/BUILD_INFO 2>/dev/null');
  console.log('Admin BUILD: ' + (r || 'NONE'));

  r = await execCmd('docker exec 6b099ed3-7175-4a78-91f4-44570c84ed27-h5 cat /app/BUILD_INFO 2>/dev/null');
  console.log('H5 BUILD: ' + (r || 'NONE'));

  conn.end();
  console.log('DONE');
}
main().catch(function(e) { console.error('ERR: ' + e); process.exit(1); });

File: deploy/check_server.py
"""Check server state and save to file."""
import paramiko, sys

ssh = paramiko.SSHClient()
ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
ssh.connect('newbb.test.bangbangvip.com', port=22, username='ubuntu', password='Newbang888', timeout=20)

results = {}

stdin, stdout, stderr = ssh.exec_command('cat /home/ubuntu/gateway/nginx.conf', timeout=10)
results['nginx_conf'] = stdout.read().decode()

stdin, stdout, stderr = ssh.exec_command('ls -la /home/ubuntu/gateway/conf.d/', timeout=10)
results['confd'] = stdout.read().decode()

stdin, stdout, stderr = ssh.exec_command('ls /home/ubuntu/6b099ed3-7175-4a78-91f4-44570c84ed27/ 2>&1', timeout=10)
results['proj_dir'] = stdout.read().decode()

stdin, stdout, stderr = ssh.exec_command('docker ps -a --filter name=6b099ed3 --format "{{.Names}} {{.Status}}" 2>&1', timeout=10)
results['containers'] = stdout.read().decode()

ssh.close()

with open('C:/auto_output/bnbbaijkgj/server_state.txt', 'w', encoding='utf-8') as f:
    for k, v in results.items():
        f.write(f'=== {k} ===\n{v}\n\n')
print('SAVED to server_state.txt')
